### PR TITLE
Setup DB, Redis, security utils & CI

### PR DESCRIPTION
**Pull request by ahsanhabib9191**
# DB & Middleware: Mongo/Redis setup, encryption, auth, rate limiting, tests, and migration

This PR introduces the database and middleware foundation for the Meta ads optimization stack.

Highlights

- Database & Infra
  - MongoDB/Mongoose with index drop+sync on init, connection caching, Docker Compose for MongoDB 6 and Redis 7
  - Redis client for rate limiting
- Utilities & Security
  - AES-256-GCM encryption for Meta tokens (encrypt on save/update)
  - Winston logger with daily rotation; Zod validators; Meta scopes helpers
- Middleware
  - Auth: JWT verification (NEXTAUTH_SECRET), API key verification via Tenant.apiKeyHash, requireAuth wrapper
  - Rate limiting: Redis sliding window by IP/tenant/api key
  - Central error handler
- Models
  - MetaConnection: encrypted tokens, status field, indexes
  - Tenant: apiKeyHash, requestCounts, issue/verify API key helpers
- Migration & Tests
  - scripts/encrypt-existing-tokens.ts (dry vs live)
  - scripts/test-encryption.ts (encryption + API key lifecycle)
  - scripts/test-auth.ts (verifyAuth, requireAuth, verifyApiKey)
  - scripts/test-rate-limit.ts (window behavior, edge cases)
- README & Scripts
  - Docs for migration and tests; npm scripts: test:security, test:auth, test:rate

How to run

1) npm install
2) npm run docker:up
3) export NEXTAUTH_SECRET="dev-secret"
4) export ENCRYPTION_KEY="$(node -e "crypto=require('crypto');console.log(crypto.randomBytes(32).toString('hex'))")"
5) npm run test:security && npm run test:auth && npm run test:rate

Status: Build + typecheck green; tests pass locally.
Follow-ups: add CI workflow and integrate with Next.js API routes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+          cache: 'npm'
+      - run: |
+          if [ -f package-lock.json ]; then npm ci; else npm install; fi
+      - name: Wait for services
+        run: |
+          for i in {1..30}; do
+            (mongosh --eval 'db.runCommand({ ping: 1 })' && redis-cli ping) && exit 0 || sleep 2;
```
[the remark this answers is not in the document]
Comment (ahsanhabib9191) [reply .github/workflows/ci.yml]:
```suggestion
            (mongosh --eval 'db.runCommand({ ping: 1 })' && redis-cli ping) && exit 0 || sleep 2;
```

[the remark this answers is not in the document]
Comment (ahsanhabib9191) [reply .github/workflows/ci.yml]:
```suggestion
            (mongosh --eval 'db.runCommand({ ping: 1 })' && redis-cli ping) && exit 0 || sleep 2;
```

File: lib/utils/validators.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+export const emailSchema = z.string().email({ message: 'Invalid email address' });
+export function validateEmail(email: string): boolean {
+  return emailSchema.safeParse(email).success;
+}
+
+export const metaAdAccountIdSchema = z.string().regex(/^act_\d{6,}$/,{ message: 'Invalid Meta Ad Account ID (expected act_XXXXXXXXXX)' });
+export const metaPixelIdSchema = z.string().regex(/^\d+$/,{ message: 'Invalid Meta Pixel ID (numeric string expected)' });
+export const metaPageIdSchema = z.string().regex(/^\d+$/,{ message: 'Invalid Meta Page ID (numeric string expected)' });
```
[the remark this answers is not in the document]
Comment (ahsanhabib9191) [reply lib/utils/validators.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ahsanhabib9191/meta-ad/pull/1#discussion_r2591250414) fix this

File: lib/middleware/rate-limit.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+const redis = new Redis(process.env.REDIS_URL || 'redis://localhost:6379', { lazyConnect: true });
+
+export function createRateLimiter(options: RateLimitOptions) {
+  const { windowMs, maxRequests, keyPrefix } = options;
+  return async function (key: string): Promise<RateLimitResult> {
+    const now = Date.now();
+    const windowStart = now - windowMs;
+    const zkey = `${keyPrefix}:${key}`;
```
[the remark this answers is not in the document]
Comment (ahsanhabib9191) [reply lib/middleware/rate-limit.ts]: @copilot fix this
